indico/web/forms/fields.py

```diff
@@ -351,6 +360,12 @@ def _get_event_person(self, data):
 
     def _serialize_principal(self, principal):
         from indico.modules.events.util import serialize_event_person
+        if principal.id is None:
+            # We created an EventPerson which has not been persisted to the
+            # database. Revert the conversion.
+            principal = self.event_person_conversions[principal]
+            if isinstance(principal, dict):
```

Comment:
Yes, it could be a User (see line 339) in which case the method of the superclass will handle it.